prov/efa/src/efa_hmem.c

```diff
@@ -283,20 +282,18 @@ efa_domain_hmem_info_init_iface(struct efa_domain *efa_domain, enum fi_hmem_ifac
 	* memory, therefore p2p is not required.
 	*/
 	if (iface == FI_HMEM_CUDA &&
-	    FI_VERSION_GE(efa_domain->util_domain.fabric->fabric_fid.api_version, FI_VERSION(1, 18)))
+	    FI_VERSION_GE(g_device_list[0].rdm_info->fabric_attr->api_version, FI_VERSION(1, 18)))
```

Comment:
Before your change, the `api_version` is the one passed from `fi_getinfo(FI_VERSION(...))`, after your change, I think it is the api version of the current libfabric library (please double check on this). They are different so I think you'd need to move `p2p_required_by_impl` from this struct too.

Reply:
We only have conditional p2p_required_by_impl for cuda, I think it's easy to just define this within the p2p check in ep.